evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
I'm not convinced.  The original behavior was, if ``StopProcessing`` is raised, return whatever the ``retval`` is…and ``retval`` defaults to ``None``.  The code I have that uses evproc in fact does a ``raise evproc.StopProcessing()`` and expects all event processing to halt.  Of course, those processors aren't returning any additional events, as they would use ``proc.process()`` to accomplish the same thing…